cluster/src/main/java/org/apache/iotdb/cluster/log/manage/RaftLogManager.java

```diff
@@ -415,6 +423,9 @@ public void commitTo(long newCommitIndex, boolean ignoreExecutionExceptions)
               .clear();
         }
         try {
+          if (committedEntryManager.getTotalSize() + entries.size() > maxNumOfLogsInMem) {
+            throw new LogNumberOutOfBoundException();
+          }
```

Comment:
You should trigger a log deletion and proceed the committing instead of throwing an exception.

Reply:
I see. Thanks for your suggestion.